src/main/java/com/duckblade/osrs/toa/features/tomb/DryStreakTracker.java

```diff
@@ -0,0 +1,62 @@
+package com.duckblade.osrs.toa.features.tomb;
+
+import com.duckblade.osrs.toa.TombsOfAmascutConfig;
+import com.duckblade.osrs.toa.module.PluginLifecycleComponent;
+import com.duckblade.osrs.toa.util.RaidRoom;
+import com.duckblade.osrs.toa.util.RaidState;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.Client;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.chat.ChatMessageManager;
+import net.runelite.client.chat.QueuedMessage;
+
+@Slf4j
+@Singleton
+public class DryStreakTracker implements PluginLifecycleComponent
+{
+	private static final int VARBIT_ID_SARCOPHAGUS = 14373;
+
+	@Inject
+	private Client client;
+	@Inject
+	private ClientThread clientThread;
+	@Inject
+	private TombsOfAmascutConfig config;
+	@Inject
+	private ChatMessageManager chatMessageManager;
+
+	@Override
+	public boolean isEnabled(final TombsOfAmascutConfig config, final RaidState raidState)
+	{
+		return raidState.getCurrentRoom() == RaidRoom.TOMB && config.trackPurpleDryCount();
+	}
+
+	@Override
+	public void startUp()
+	{
+		clientThread.invokeLater(() ->
+		{
+			final boolean purple = client.getVarbitValue(VARBIT_ID_SARCOPHAGUS) % 2 != 0;
+
+			final int dryStreak = config.getPurpleDryStreakCount() + 1;
+
+			final String msg = String.format("Purple Dry Streak: <col=ff0000>%d</col>", dryStreak);
+
+			chatMessageManager.queue(QueuedMessage.builder()
+				.type(ChatMessageType.GAMEMESSAGE)
+				.runeLiteFormattedMessage(msg)
+				.build());
+
+			config.setPurpleDryStreakCount(purple ? 0 : dryStreak);
```

Comment:
This is currently going to appear whether the user has pulled a purple or not, which I think is misleading if they do pull one. But also, I wouldn't want it to just not appear if they get a purple, since then it can give it away.

(Also, right now toggling the feature would increment the count arbitrarily, but in order to not it requires a bit of jank.)

Still increment the counter in `startUp`, we'll ignore the enable-spam edge case for now, and then display the message (ideally only once?) upon opening the loot chest. You'll also need to make it work in the lobby for users claiming outside, but make sure you check the raid room before incrementing so you don't do so in the lobby.

Reply:
![java_T1H9rpjGKz](https://github.com/LlemonDuck/tombs-of-amascut/assets/38548565/2e8abaa1-ef46-4c00-9566-bd4d43093436)

This is the behavior with the latest commit. The downside to this is that the counter is reset before showing the message, so you won't know what count you broke your dry streak on. Not showing the message at all might also be misleading :/